src/CodeExampleLanguageProcessing.ts

```diff
@@ -0,0 +1,371 @@
+  },
+  {
+    directiveValue: "json",
+    canonicalValue: CanonicalLanguageValues.JSON,
+    extension: LanguageFileExtensions.JSON,
+  },
+  {
+    directiveValue: "JSON",
+    canonicalValue: CanonicalLanguageValues.JSON,
+    extension: LanguageFileExtensions.JSON,
+  },
+  {
+    directiveValue: "kotlin",
+    canonicalValue: CanonicalLanguageValues.KOTLIN,
+    extension: LanguageFileExtensions.KOTLIN,
+  },
+  {
+    directiveValue: "none",
+    canonicalValue: CanonicalLanguageValues.UNDEFINED,
+    extension: LanguageFileExtensions.UNDEFINED,
+  },
+  {
+    directiveValue: "php",
+    canonicalValue: CanonicalLanguageValues.PHP,
+    extension: LanguageFileExtensions.PHP,
+  },
+  {
+    directiveValue: "python",
+    canonicalValue: CanonicalLanguageValues.PYTHON,
+    extension: LanguageFileExtensions.PYTHON,
+  },
+  {
+    directiveValue: "ruby",
+    canonicalValue: CanonicalLanguageValues.RUBY,
+    extension: LanguageFileExtensions.RUBY,
+  },
+  {
+    directiveValue: "rust",
+    canonicalValue: CanonicalLanguageValues.RUST,
+    extension: LanguageFileExtensions.RUST,
+  },
+  {
+    directiveValue: "scala",
+    canonicalValue: CanonicalLanguageValues.SCALA,
+    extension: LanguageFileExtensions.SCALA,
+  },
+  {
+    directiveValue: "sh",
+    canonicalValue: CanonicalLanguageValues.SHELL,
+    extension: LanguageFileExtensions.SHELL,
+  },
+  {
+    directiveValue: "shell",
+    canonicalValue: CanonicalLanguageValues.SHELL,
+    extension: LanguageFileExtensions.SHELL,
+  },
+  {
+    directiveValue: "swift",
+    canonicalValue: CanonicalLanguageValues.SWIFT,
+    extension: LanguageFileExtensions.SWIFT,
+  },
+  {
+    directiveValue: "text",
+    canonicalValue: CanonicalLanguageValues.TEXT,
+    extension: LanguageFileExtensions.TEXT,
+  },
+  {
+    directiveValue: "typescript",
+    canonicalValue: CanonicalLanguageValues.TYPESCRIPT,
+    extension: LanguageFileExtensions.TYPESCRIPT,
+  },
+  {
+    directiveValue: "undefined",
+    canonicalValue: CanonicalLanguageValues.UNDEFINED,
+    extension: LanguageFileExtensions.UNDEFINED,
+  },
+  {
+    directiveValue: "xml",
+    canonicalValue: CanonicalLanguageValues.XML,
+    extension: LanguageFileExtensions.XML,
+  },
+  {
+    directiveValue: "yaml",
+    canonicalValue: CanonicalLanguageValues.YAML,
+    extension: LanguageFileExtensions.YAML,
+  },
+  {
+    directiveValue: "json\\n :copyable: false",
+    canonicalValue: CanonicalLanguageValues.JSON,
+    extension: ".json",
+  },
+  {
+    directiveValue: "json\\n:copyable: true",
+    canonicalValue: CanonicalLanguageValues.JSON,
+    extension: ".json",
+  },
+];
+
+export const getLanguageDetailsForCodeNode = (
+  node: CodeNode
+): LanguageMapper => {
+  if (node.lang != null) {
+    const mapping = languageValueMappings.find(
+      (mapper) => mapper.directiveValue === node.lang
+    );
+    if (mapping != undefined) {
+      return mapping;
+    } else {
```

Comment:
No need for else after return.
If you do use early return style (as I do), prefer error/unhappy case return first.
In this case it would be 
```js
if (mapping === undefined) {
  return { ... undefined ... };
}
return mapping;
```
Likewise - the check above (`node.lang == null`) would exit early and avoid indenting and nesting all these ifs.